services/QuillLMS/app/services/canvas_integration/user_importer.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module CanvasIntegration
+  class UserImporter < ApplicationService
+    attr_reader :email, :external_id, :name, :role, :url
+
+    APPROVED_ROLES = [User::TEACHER, User::STUDENT].freeze
+    DEFAULT_ROLE = User::STUDENT
+
+    def initialize(email:, external_id:, name:, role:, url:)
+      @email = email
+      @external_id = external_id
+      @name = name
+      @role = role
+      @url = url
+    end
+
+    def run
+      existing_canvas_account_user || new_canvas_account_user
+    end
+
+    private def canvas_instance
+      @canvas_instance ||= CanvasInstance.find_by!(url: url)
+    end
+
+    private def existing_canvas_account_user
+      CanvasAccount
+        .find_by(external_id: external_id, canvas_instance: canvas_instance)
+        &.user
+    end
+
+    private def new_canvas_account_user
+      user
+        .canvas_accounts
+        .build(canvas_instance: canvas_instance, external_id: external_id)
+        .tap(&:save!)
```

Comment:
I think this can be shortened
```suggestion
        .create!(canvas_instance: canvas_instance, external_id: external_id)
```

Reply:
I'm using `build` here to [satisfy](https://github.com/empirical-org/Empirical-Core/blob/e7f553406423091b9c1c368c74098a9916dd5d3f/services/QuillLMS/app/models/user.rb#L872) the `requires_password` validation on User.
The `tap(&:save!)` returns saved object (i.e. the newly created canvas account) so that a subsequent call to `.user` can be made.

Reply:
But you're right to flag this.  I added a comment on why build (and not create) is being used here.